File: Ridy-taxi-app/ios/common/Models/Objects/PaymentGateway.swift
//
//  PaymentGateway.swift
//  Shared
//
//  Created by Manly Man on 12/11/19.
//  Copyright © 2019 Innomalist. All rights reserved.
//

import UIKit

public class PaymentGateway: Codable, Equatable {
    public static func == (lhs: PaymentGateway, rhs: PaymentGateway) -> Bool {
        return lhs.id == rhs.id
    }
    
    public var id: Int
    var title: String
    public var type: PaymentGatewayType
    public var publicKey: String?
}

public enum PaymentGatewayType: String, Codable {
    case Stripe = "stripe"
    case BrainTree = "braintree"
    case PayPal = "paypal"
    case Paytm = "paytm"
    case Razorpay = "razorpay"
    case Paystack = "paystack"
    case PayU = "payu"
    case Instamojo = "instamojo"
    case Flutterwave = "flutterwave"
    case PayGate = "paygate"
    case MIPS = "mips"
    case CustomLink = "link"
}

File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/models/PaymentGateway.kt
package com.innomalist.taxi.common.models

import com.squareup.moshi.Json

data class PaymentGateway(
    var id: Int,
    var title: String?,
    var type: PaymentGatewayType,
    var publicKey: String? = null,
    var privateKey: String? = null
)
enum class PaymentGatewayType (val rawValue: String) {
    @Json(name="stripe")
    Stripe("stripe"),
    @Json(name="braintree")
    Braintree("braintree"),
    @Json(name = "paypal")
    PayPal("paypal"),
    @Json(name = "paytm")
    Paytm("paytm"),
    @Json(name = "razorpay")
    Razorpay("razorpay"),
    @Json(name = "paystack")
    Paystack("paystack"),
    @Json(name = "payu")
    PayU("payu"),
    @Json(name = "instamojo")
    Instamojo("instamojo"),
    @Json(name="flutterwave")
    Flutterwave("flutterwave"),
    @Json(name="paygate")
    PayGate("paygate"),
    @Json( name = "mips")
    MIPS("mips"),
    @Json( name = "link")
    CustomLink("link");

    companion object {
        operator fun invoke(rawValue: String) = values().firstOrNull { it.rawValue == rawValue }
    }
}
